python/dataverse_sdk/utils/utils.py

```diff
@@ -33,3 +34,39 @@ def download_file_from_response(response: requests.models.Response, save_path: s
             if chunk:
                 file.write(chunk)
                 file.flush()
+
+
+def download_file_from_url(url: str, save_path: str):
+    """Downloads a file from a URL and saves it to the specified path.
+
+    Parameters
+    ----------
+    url : str
+        The URL of the file to download.
+    save_path : str
+        The local file path where the file will be saved.
+
+    """
+    try:
+        # Send a HTTP request to the URL
+        response = requests.get(url, stream=True)
+        response.raise_for_status()  # Check if the request was successful
+
+        # Get the total file size from headers (if available)
+        total_size = int(response.headers.get("content-length", 0))
+
+        # Initialize tqdm progress bar
+        with tqdm(
+            total=total_size, unit="B", unit_scale=True, desc="Downloading"
+        ) as progress_bar:
+            # Write the file in chunks to avoid using too much memory
+            with open(save_path, "wb") as file:
+                for chunk in response.iter_content(chunk_size=8192):
+                    if chunk:  # Filter out keep-alive chunks
+                        file.write(chunk)
+                        progress_bar.update(len(chunk))  # Update progress bar
```

Comment:
Nit: can combine with statement together.
```suggestion

        with open(save_path, "wb") as file, tqdm(
            total=total_size,
            unit="B",
            unit_scale=True,
            desc=f"Downloading",
        ) as progress_bar:
            for chunk in response.iter_content(chunk_size=8192):
                if chunk:
                    file.write(chunk)
                    progress_bar.update(len(chunk))
```

Reply:
updated, thanks!